core/src/main/java/org/elasticsearch/action/fieldstats/FieldStats.java

```diff
@@ -449,7 +450,11 @@ public String getMaxValue() {
 
         @Override
         protected java.lang.Long valueOf(String value) {
-            return dateFormatter.parser().parseMillis(value);
+            try {
+                return dateFormatter.parser().parseMillis(value);
+            } catch (Exception e) {
```

Comment:
Ok, I'll remove the fallback logic and add a `format` option to the api that overrides the format that has been specified in the mapping for a date field.